tree/treeplayer/src/TDFUtils.cxx

```diff
@@ -152,6 +153,41 @@ unsigned int GetNSlots()
    return nSlots;
 }
 
+void GetBranchNamesImpl(TTree* t, std::set<std::string> &bNames, std::set<TTree*> &analysedTrees) {
+
+   if (!t) return;
+
+   if (! analysedTrees.insert(t).second) {
+      return;
+   }
+
+   auto branches = t->GetListOfBranches();
+   if (branches) {
+      for (auto branchObj : *branches) {
+         bNames.insert(branchObj->GetName());
+      }
+   }
+
+   auto friendTrees = t->GetListOfFriends();
+
+   if (!friendTrees) return;
+
+   for (auto friendTreeObj : *friendTrees) {
+      auto friendTree = ((TFriendElement*) friendTreeObj)->GetTree();
+      GetBranchNamesImpl(friendTree, bNames, analysedTrees);
+   }
+}
+
+ColumnNames_t GetBranchNames(TTree* t)
```

Comment:
I think this function should take a reference to a TTree and we should not call it if there is no tree..